### PR TITLE
[Puurfect-ui] Add table

### PR DESCRIPTION
**Pull request by MarcoPOtard**

### CODE REVIEW

**Comment by MatouTechnologies** [comment]
This pull request has been deployed to Vercel.
<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://matou-technologies-purrfect-ui-storybook-37.vercel.app'>https://matou-technologies-purrfect-ui-storybook-37.vercel.app</a></td>
  </tr>
  <tr>
    <td><strong>🔍 Inspect:</strong></td>
    <td><a href='https://vercel.com/matou-technologies/purrfect-ui/GvtvXupHE55o1sqoMp9VFg6d52Kq'>https://vercel.com/matou-technologies/purrfect-ui/GvtvXupHE55o1sqoMp9VFg6d52Kq</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="deploy-preview-storybook"></hidden>

File: src/display/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  position: relative;
+  border: 1px solid ${theme.color.gray100};
+  border-radius: ${theme.borderRadius.default};
+  background: ${theme.color.white};
+`;
+
+const StyledTable = styled(AriaTable)`
+  ${body1};
+  width: 100%;
+  table-layout: initial !important;
```
Comment (Rcaradec) [inline src/display/Table/Table.tsx:47]: can we avoid using **!important** ?

File: src/display/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            <TableRow
+              columns={props.dataColumns}
+              key={row.id}
+              isDisabled={row.isDisabled}
+              data-is-colored={row.isColored}
+            >
+              {props.dataColumns.map((column) => {
+                return (
+                  <TableCell key={column.id}>
+                    {(row as Record<string, any>)[column.id ?? "name"]}
```
Comment (Rcaradec) [inline src/display/Table/Table.tsx:156]: maybe we can find better than **any** here

File: src/display/Table/Table.tsx (hunk 1 above)
Comment (MarcoPOtard) [reply src/display/Table/Table.tsx:48]: I didn't find solution to overload the "table-layout" is setting in the style attribute with js

File: src/display/Table/Table.tsx (hunk 2 above)
Comment (MarcoPOtard) [reply src/display/Table/Table.tsx]: Record need two arguments, the first one is what we want and the second one is what is it.

File: src/display/Table/Table.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      />,
+    );
+
+    const row = screen.getByRole("row", { name: /Games/i });
+    await waitFor(() => userEvent.click(row));
+
+    expect(row).toHaveAttribute("aria-selected", "true");
+  });
+
+  it("Should click only on one row", async () => {
```
Comment (Rcaradec) [inline src/display/Table/Table.spec.tsx:138]: "Should allow only one selected row"  ?

Comment (MarcoPOtard) [reply src/display/Table/Table.spec.tsx:138]: Changed

File: src/form/Checkbox/Checkbox.stories.tsx [hunk 4]
```diff
@@ -75,3 +81,26 @@ export const checkboxDisabled: Story = {
...
+
+export const checkboxIndeterminate: Story = {
+  args: {
+    label: "Meow",
+    isIndeterminate: true,
+  },
+  render: (args) => (
+    <>
+      <Title>Checkbox is Indeterminate</Title>
+      <div
```
[the remark this answers is not in the document]
Comment (MarcoPOtard) [reply src/form/Checkbox/Checkbox.stories.tsx:93]: oh merde. Fixed

**Review by Rcaradec — APPROVED** [review]

**Pull request merged**